packages/astro/src/core/config/settings.ts

```diff
@@ -134,3 +145,63 @@ export async function createSettings(config: AstroConfig, cwd?: string): Promise
 
 	return settings;
 }
+
+function validateTsconfig(settings: AstroSettings, logger: Logger, rawConfig: TSConfig) {
+	try {
+		if (!settings.config.experimental.tsconfig) {
+			return;
+		}
+
+		let newConfig = { ...rawConfig };
+
+		if (!rawConfig.extends) {
+			newConfig.extends = [GENERATED_TSCONFIG_PATH];
```

Comment:
The path to the generate tsconfig path isn't always `./.astro/tsconfig.json` if the user has a different root. If we want this, we'd need to take in account of that. Related to https://github.com/withastro/astro/pull/11832#discussion_r1732952227